### PR TITLE
[complex32, jiterator] sin, asin

### PR DESCRIPTION
**Pull request by khushi-411**
Follows #74537 and #74748

cc @kshitij12345

### CODE REVIEW

**Review by kshitij12345 — COMMENTED** [review]
Looks good!

Just need to take care of CI error with expectedFailure.

https://github.com/pytorch/pytorch/runs/6465060483?check_suite_focus=true#step:9:10473

Thanks @khushi-411 !

**Comment by khushi-411** [comment]
Thanks, @kshitij12345! I got the point for that failure. I think it will work now :)

**Review by kshitij12345 — APPROVED** [review]
LGTM! Thanks @khushi-411 

cc: @anjali411

**Comment by khushi-411** [comment]
Hi, @anjali411! Gentle ping :)

**Comment by khushi-411** [comment]
Hi, @anjali411! Gentle ping :)

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -18959,14 +18960,14 @@ def __init__(
     ElementwiseUnaryPythonRefInfo(
         "_refs.sin",
         torch_opinfo_name="sin",
-        skips=(
+        #skips=(
```
Comment (ngimel) [inline torch/testing/_internal/common_methods_invocations.py:18963]: please actually remove skip, not just comment out

Comment (khushi-411) [reply torch/testing/_internal/common_methods_invocations.py]: Oops, I actually forgot to remove it. Removed that now. Thanks!

File: aten/src/ATen/native/cuda/UnaryGeometricKernels.cu [hunk 2]
```diff
@@ -21,15 +22,42 @@ void acos_kernel_cuda(TensorIteratorBase& iter) {
       });
 }
 
+const char asin_name[] = "asin";
 void asin_kernel_cuda(TensorIteratorBase& iter) {
-  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(
+auto common_dtype = iter.common_dtype();
```
Comment (kshitij12345) [inline aten/src/ATen/native/cuda/UnaryGeometricKernels.cu:27]: minor nit: this is under-indented.

Comment (khushi-411) [reply aten/src/ATen/native/cuda/UnaryGeometricKernels.cu]: Ah, right! Thanks, @kshitij12345! Have updated it :)

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**